mwdb/web/src/components/UserPasswordRecover.js

```diff
@@ -1,124 +1,111 @@
-import React, { Component } from "react";
+import React, { useState, useContext } from "react";
 import ReCAPTCHA from "react-google-recaptcha";
 
 import api from "@mwdb-web/commons/api";
 import { ConfigContext } from "@mwdb-web/commons/config";
 import { View } from "@mwdb-web/commons/ui";
 
-export default class UserPasswordRecover extends Component {
-    constructor(props) {
-        super(props);
-        this.state = this.initialState;
-    }
-
-    static contextType = ConfigContext;
-
-    get initialState() {
-        return {
-            login: "",
-            email: "",
-            success: false,
-            dirty: false,
-            error: null,
-            recaptcha: null,
-        };
-    }
+export default function UserPasswordRecover() {
+    const initialState = {
+        login: "",
+        email: "",
+        success: false,
+        error: null,
+        recaptcha: null,
+    };
 
-    componentDidMount() {
-        if (this.dirty) this.setState(this.initialState);
-    }
+    const config = useContext(ConfigContext);
+    const [state, setState] = useState(initialState);
```

Comment:
Maybe it would be better to have separate state variables for these fields? :thinking: